File: Python/calculator.py
# A module containing the code logic

def add(x, y):
    """
    This function computes the addition of two numbers.

    Args:
        x: first number
        y: second number

    Returns:
        The addition of the two numbers.

    """
    return x + y

def subtract(x, y):
    """
    This function computes the subtraction of two numbers.

    Args:
        x: first operand
        y: second operand

    Returns:
        The subtraction of the two operands

    """
    return x - y

def multiply(x, y):
    """
    This function computes the multiplication of two operands.

    Args:
        x: first operand
        y: second operand

    Returns:
        The multiplication of the two numbers

    """
    return x * y

def divide(x, y):
    """
    Computes the division of two numbers

    Args:
        x: first number
        y: second number

    Returns:
        The division of the numbers
    """
    
    # if the denominator is zero(0)
    if y == 0:
        return "Error! Division by zero."
    else:
        return x / y

def calculator():
    """
    This the main function the allows a user
    perform the calculator's basic operation.
    """

    print("Select operation:")
    print("A: Addition")
    print("S: Subtraction")
    print("M: Multiplication")
    print("D: Division")

    # user's operation input
    choice = input("Enter choice (A/S/M/D): ")

    if choice in ['A', 'S', 'M', 'D']:
        num1 = float(input("Enter first number: "))
        num2 = float(input("Enter second number: "))

        if choice == 'A':
            print(f"{num1} + {num2} = {add(num1, num2)}")

        if choice == 'S':
            print(f"{num1} - {num2} = {sub(num1, num2)}")

        if choice == 'M':
            print(f"{num1} * {num2} = {mul(num1, num2)}")

        if choice == 'D':
            print(f"{num1} / {num2} = {div(num1, num2)}")

    else:
        print("Invalid Input")

if __name__ == "__main__":
    calculator()

File: C/calculator.c
#include <stdio.h>
#include <stdlib.h>
#include "calculator.h"

/**
 * add - prints the addition of two numbers
 * @a: first number
 * @b: second number
 * Returns: a + b
 */
double add(double a, double b)
{
	return a + b;
}

/**
 * subtract - prints the subtraction of two numbers
 * @a: first number
 * @b: second number
 * Returns: a - b
 */
double subtract(double a, double b)
{
	return a - b;
}

/**
 * multiply - prints the multiplication of two numbers
 * @a: first number
 * @b: second number
 * Returns: a * b
 */
double multiply(double a, double b)
{
	return a * b;
}

/**
 * divide - prints the division of two numbers to STDOUT
 * @a: first number
 * @b: second number
 * Returns: a / b if b != 0 else -1
 */
double divide(double a, double b)
{
	if (b == 0)
		return -1;
	else
		return a / b;
}


/**
 * main - Entry point
 * Return: Always zero

int main(void)
{
	double num1, num2;
	char operation;

	printf("Enter first number: ");
	if (scanf("%lf", &num1) != 1)
	{
		printf("Invalid input. Please enter a number.\n");
		return (1);
	}

	printf("Enter an operation (+, -, *, /): ");
	scanf(" %c", &operation);

	printf("Enter second number: ");
	if (scanf("%lf", &num2) != 1)
	{
		printf("Invalid input. Please enter a number.\n");
		return (1);
	}

	switch (operation)
	{
		case '+':
			add(num1, num2);
			break;

		case '-':
			subtract(num1, num2);
			break;

		case '*':
			multiply(num1, num2);
			break;

		case '/':
			divide(num1, num2);
			break;

		default:
			printf("Error: Invalid operation.\n");
			return (1);
	}

	return (0);
}
*/
